distribution/push.go

```diff
@@ -127,6 +133,19 @@ func Push(ctx context.Context, ref reference.Named, imagePushConfig *ImagePushCo
 			continue
 		}
 
+		parsedURL, urlErr := url.Parse(endpoint.URL)
```

Comment:
can we extract this logic to a function? it looks like it's duplicated in `pull.go`.